core/primitives/src/lib.rs

```diff
@@ -286,3 +286,25 @@ impl From<LogLevel> for log::Level {
 		}
 	}
 }
+
+/// Converts the given value into an `u64` by following the Substrate calling convention.
+///
+/// When Substrate calls into Wasm it expects a fixed signature for functions exported
+/// from the Wasm blob. The return value of this signature is always a `u64` in little endian.
+/// This `u64` stores the pointer to the encoded return value and the length of this encoded value.
+/// The first `32bit` are reserved for the pointer and the last `32bit` for the length.
```

Comment:
This on the contrary depends on the bit numbering scheme: the first would mean "the rightmost" bit in LSB and "the leftmost" in MSB. I'd go for "high/low" parts.